nasa/ow_simulator
# Oceanwater 1010 implement telemetry arm faults status

## Linked Issues:
| EPIC ⚡ | [OCEANWATER-934](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-934) |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-1010](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1010) |
| Github :octocat:  | [OCEANWATER-1010_implement_telemetry_arm_faults_status](https://github.com/nasa/ow_simulator/tree/OCEANWATER-1010_implement_telemetry_arm_faults_status) |

Mostly artifact renaming and updating message definitions.

**This issue has a corresponding and required branch on [ow_autonomy](https://github.com/nasa/ow_autonomy/tree/OCEANWATER-1010_implement_telemetry_arm_faults_status).**

## Summary of Changes
The largest change, present in all tickets within the epic, is the creation of the `owl_msgs` package. The new `owl_msgs` package houses telemetry messages used by the fault detection and injection packages, according to the OWLAT-OceanWATERS Telemetry Unification plan. 

* Created the `owl_msgs` package.
* Updated `ow_faults_detection` to depend on `owl_msgs`

Specific to this ticket:
* Renamed the `ArmFaults` message to `ArmFaultsStatus` and updated all references in source and documentation.
* Moved `ArmFaultsStatus.msg` to the new `owl_msgs` package.
* Removed the `/faults/` topic prefix from the `arm_faults_status` topic.
* Updated `ArmFaultsStatus` message semantics and defintion.
* Made Plexil-OwInterface package imports consistent with new 'owl_msgs' package.
* Renamed some variables to be consistent with message rename.

## Test
* Build according to [User Documentation](https://github.com/nasa/ow_simulator/wiki/Using-OceanWATERS).
* Run two terminals sourced with the project's `setup.bash`.
* In one terminal, run the project:
  * `roslaunch ow atacama_y1a.launch`
* In the second terminal, verify the fault messages are in the `owl_msgs` package and its contents:
  * `rosmsg package owl_msgs`
  * `rosmsg show owl_msgs/ArmFaultsStatus`
* In the second terminal Subscribe to the system faults status topic (see *Comments/Issues*)
  * `rostopic echo /system_faults_status`
* In RQT go to the Dynamic Reconfigure tab and select faults.
* Toggle any of the arm faults and observe the rostopic change with an appropriate value

## Comments/Issues
* Individual arm faults are not currently implemented. Only visible in the encapsulating `system_faults_status` "ARM_EXECUTION_ERROR" error.
